encrypt/encrypt.go

```diff
@@ -27,14 +36,271 @@ var ArgHelp = `
 
 var Args = flag.NewFlagSet("encrypt", flag.ExitOnError)
 
+var publicKeyFile = Args.String("key", "",
+	"Public key to use for encrypting files.")
+var outDir = Args.String("outdir", "", "Output directory for encrypted files")
+
+// Type to keep track of infiles and outfiles
+type encryptionFileSet struct {
+	unencrypted string
+	encrypted   string
+}
+
+type hashSet struct {
+	encryptedMd5      string
+	unencryptedMd5    string
+	encryptedSha256   string
+	unencryptedSha256 string
+}
+
 // Main encryption function
 func Encrypt(args []string) {
+	log.Infof("args: %s", strings.Join(os.Args[:], ", "))
 	// Parse flags. There are no flags at the moment, but in case some are added
 	// we check for them.
 	Args.Parse(os.Args[1:])
 
 	// Args() returns the non-flag arguments, which we assume are filenames.
-	files := Args.Args()
+	// All filenames are read into a struct together with their output filenames
+	files := []encryptionFileSet{}
+	for _, filename := range Args.Args() {
+
+		// Set directory for the output file
+		outFilename := filename + ".c4gh"
+		if *outDir != "" {
+			_, basename := path.Split(filename)
+			outFilename = path.Join(*outDir, basename) + ".c4gh"
+		}
+
+		files = append(files, encryptionFileSet{filename, outFilename})
+	}
+
+	// Check that all the infiles exist, and all the outfiles don't
+	err := checkFiles(files)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
You are pulling in Logrus, so make use of it and generate proper verbose log messages

Reply:
yes!